libs/saml2-web-sso/src/SAML2/WebSSO/SP.hs

```diff
@@ -272,33 +281,56 @@ instance (Monad m, SPStoreID i m) => SPStoreID i (JudgeT m) where
 --  * 'astSubjectLocality' ("This element is entirely advisory, since both of these fields are
 --    quite easily “spoofed,” but may be useful information in some applications." [1/2.7.2.1])
 --
--- 'judge' does check the 'rspStatus' field, even though that makes no sense: most IdPs encrypt the
--- 'Assertion's, but not the entire response, and we make taht an assumption when validating
--- signatures.  So the status info is not signed, and could easily be changed by an attacker
--- attempting to authenticate.
-judge :: (Monad m, SP m, SPStore m) => AuthnResponse -> JudgeCtx -> m AccessVerdict
-judge resp ctx = runJudgeT ctx (judge' resp)
-
-judge' :: (HasCallStack, MonadJudge m, SP m, SPStore m) => AuthnResponse -> m AccessVerdict
-judge' resp = do
-  case resp ^. rspStatus of
-    StatusSuccess -> pure ()
-    StatusFailure -> deny DeniedStatusFailure
-  uref <- either (giveup . DeniedBadUserRefs) pure $ getUserRef resp
-  inRespTo <- either (giveup . DeniedBadInResponseTos) pure $ rspInResponseTo resp
-  checkInResponseTo "response" inRespTo
-  checkIsInPast DeniedIssueInstantNotInPast $ resp ^. rspIssueInstant
-  maybe (pure ()) (checkDestination DeniedBadDestination) (resp ^. rspDestination)
```

Comment:
This check got lost. If we don't want to re-establish it, we should delete `DeniedBadDestination`.